packages/vm/tests/api/EIPs/eip-3074-authcall.spec.ts

```diff
@@ -0,0 +1,680 @@
+
+    await vm.stateManager.putContractCode(contractAddress, code)
+    const tx = Transaction.fromTxData({
+      to: contractAddress,
+      gasLimit: 1000000,
+      gasPrice: 10,
+    }).sign(callerPrivateKey)
+
+    const account = await vm.stateManager.getAccount(callerAddress)
+    account.balance = BigInt(10000000)
+    await vm.stateManager.putAccount(callerAddress, account)
+
+    const result = await vm.runTx({ tx, block })
+    const buf = result.execResult.returnValue
+    st.ok(buf.equals(zeros(32)), 'auth puts 0 on stack on invalid signature')
+  })
+
+  t.test('Should throw if signature s > N_DIV_2', async (st) => {
+    const vm = new VM({ common })
+    const message = Buffer.from('01', 'hex')
+    const signature = flipSignature(signMessage(message, contractAddress, privateKey))
+    const code = Buffer.concat([getAuthCode(message, signature), RETURNTOP])
+
+    await vm.stateManager.putContractCode(contractAddress, code)
+    const tx = Transaction.fromTxData({
+      to: contractAddress,
+      gasLimit: 1000000,
+      gasPrice: 10,
+    }).sign(callerPrivateKey)
+
+    const account = await vm.stateManager.getAccount(callerAddress)
+    account.balance = BigInt(10000000)
+    await vm.stateManager.putAccount(callerAddress, account)
+
+    const result = await vm.runTx({ tx, block })
+    st.equal(result.execResult.exceptionError?.error, ERROR.AUTH_INVALID_S, 'threw correct error')
+  })
+
+  t.test('Should be able to call AUTH mutliple times', async (st) => {
+    const vm = new VM({ common })
+    const message = Buffer.from('01', 'hex')
+    const signature = signMessage(message, contractAddress, privateKey)
+    const signature2 = signMessage(message, contractAddress, callerPrivateKey)
+    const code = Buffer.concat([
+      getAuthCode(message, signature),
+      getAuthCode(message, signature2),
+      RETURNTOP,
+    ])
+
+    await vm.stateManager.putContractCode(contractAddress, code)
+    const tx = Transaction.fromTxData({
+      to: contractAddress,
+      gasLimit: 1000000,
+      gasPrice: 10,
+    }).sign(callerPrivateKey)
+
+    const account = await vm.stateManager.getAccount(callerAddress)
+    account.balance = BigInt(10000000)
+    await vm.stateManager.putAccount(callerAddress, account)
+
+    const result = await vm.runTx({ tx, block })
+    const buf = result.execResult.returnValue.slice(12)
+    st.ok(buf.equals(callerAddress.buf), 'auth returned right address')
+  })
+})
+
+// Setups the environment for the VM, puts `code` at contractAddress and also puts the STORECALLER bytecode at the contractStorageAddress
+async function setupVM(code: Buffer) {
+  const vm = new VM({ common })
+  await vm.stateManager.putContractCode(contractAddress, code)
+  await vm.stateManager.putContractCode(contractStorageAddress, STORECALLER)
+  const account = await vm.stateManager.getAccount(callerAddress)
+  account.balance = PREBALANCE
+  await vm.stateManager.putAccount(callerAddress, account)
```

Comment:
instead of this pattern you can use modifyAccountFields now:

```suggestion
  await vm.stateManager.modifyAccountFields(callerAddress, { balance: PREBALANCE })
```